csrc/ir/internal_nodes.h

```diff
@@ -685,6 +680,40 @@ class TORCH_CUDA_CU_API RNGOp : public Expr {
   }
 };
 
+class TORCH_CUDA_CU_API GetRNGSeedAndOffsetFromHost : public Expr {
```

Comment:
If it is safe to pass these outputs from one segment to another then disregard this comment. But if not and we need one of these per `Kernel`, then should this go into the `kir::` namespace instead?

Reply:
I think in order to pass it from one segment to another, we must first create this op at fusion definition time, just like you suggested in https://github.com/NVIDIA/Fuser/pull/655#discussion_r1276306161. But unfortunately we can not do it right now. But I do agree putting it to the `kir` namespace makes more sense, let me update it.